runtime/src/network/deterministic.rs

```diff
@@ -43,8 +44,13 @@ impl crate::Listener for Listener {
     type Stream = Stream;
 
     async fn accept(&mut self) -> Result<(SocketAddr, Self::Sink, Self::Stream), Error> {
-        let (socket, sender, receiver) = self.listener.next().await.ok_or(Error::ReadFailed)?;
-        Ok((socket, Sink { sender }, Stream { receiver }))
+        match self.listener.next().await {
+            Some((socket, sender, receiver)) => Ok((socket, Sink { sender }, Stream { receiver })),
+            None => Err(Error::ReadFailed(IoError::new(
```

Comment:
I've tried to avoid using IoError as a backstop for string error population and feel this gets close.

I'd consider ListenerClosed as an error if this nuance is important to you.

Reply:
Yes the listener closing is still being treated as an error here as before -- but maybe I don't understand this comment?